vhd/vhd-index.c

```diff
@@ -413,9 +413,12 @@ vhd_index_add_bat(vhdi_name_t *name,
 	bat.vhd_blocks     = vhd_blocks;
 	bat.vhd_block_size = vhd_block_size;
 
-	strcpy(bat.vhd_path, name->vhd);
-	strcpy(bat.index_path, name->index);
-	strcpy(bat.file_table_path, name->files);
+	strncpy(bat.vhd_path, name->vhd, sizeof(bat.vhd_path));
+	bat.vhd_path[sizeof(bat.vhd_path) - 1] = '\0';
+	strncpy(bat.index_path, name->index, sizeof(bat.index_path));
+	bat.index_path[sizeof(bat.index_path) - 1] = '\0';
+	strncpy(bat.file_table_path, name->files, sizeof(bat.file_table_path));
```

Comment:
This is repeated so many times, would be good if there was a function that guaranteed it always correctly null terminates a string copy. I think `strlcpy` is the preferred function for that (like strncpy but always null terminates).
Although https://en.wikibooks.org/wiki/C_Programming/C_Reference/nonstandard/strlcpy#Criticism it is not part of Glibc on purpose. That mempcpy suggested there seems like a good alternative, but might be useful to wrap it in a macro so we don't need to type it out N times.

Reply:
Yeah, I was going to suggest defining a `strlcpy` utility function as well.

Reply:
I have this in a follow-up change which I'm currently testing.